src/libraries/System.Net.WebSockets.Client/src/System/Net/WebSockets/BrowserWebSockets/ReceivePayload.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.InteropServices.JavaScript;
+
+namespace System.Net.WebSockets
+{
+    internal sealed class ReceivePayload
+    {
+        private readonly byte[] _dataMessageReceived;
+        private WebSocketMessageType _messageType;
+        private int _unconsumedDataOffset;
+
+        public ReceivePayload(byte[] array, WebSocketMessageType messageType = WebSocketMessageType.Binary)
+        {
+            _dataMessageReceived = array;
+            _messageType = messageType;
+        }
+
+        public ReceivePayload(ArrayBuffer arrayBuffer, WebSocketMessageType messageType = WebSocketMessageType.Binary)
+        {
+            using (var bin = new Uint8Array(arrayBuffer))
+            {
+                _dataMessageReceived = bin.ToArray();
+                _messageType = messageType;
+            }
+        }
+
+        public ReceivePayload(ArraySegment<byte> payload, WebSocketMessageType messageType = WebSocketMessageType.Binary)
+        {
+            _dataMessageReceived = payload.Array ?? Array.Empty<byte>();
+            _messageType = messageType;
+        }
+
+        public bool BufferPayload(ArraySegment<byte> arraySegment, out WebSocketReceiveResult receiveResult)
+        {
+            var bytesTransferred = Math.Min(_dataMessageReceived.Length - _unconsumedDataOffset, arraySegment.Count);
+            var endOfMessage = (_dataMessageReceived.Length - _unconsumedDataOffset) <= arraySegment.Count;
+
+            if (arraySegment.Array != null)
```

Comment:
It is because of `error CS8604: Possible null reference argument for parameter`

Reply:
Yes, but this is only used internally.  Most of the callers are actually constructing the ArraySegment and are doing so in a way where it'll never be null.  For the WebSocket.ReceiveAsync public entrypoint method, that method needs to validate the ArraySegment with https://github.com/dotnet/runtime/blob/a7f2f43ec3cad3c5df596cedea6ab5cb221e73e5/src/libraries/Common/src/System/Net/WebSockets/WebSocketValidate.cs#L120, at which point this also won't be null.  So rather than doing a runtime check, we should just either Debug.Assert that it's not null, or use `!` on the Array, both of which will silence the warning.

Reply:
Addressed in recent commits